### PR TITLE
Add ConvMixer Models

### PR DESCRIPTION
**Pull request by IMvision12**
# What does this PR do?
 Based on this discussion : https://github.com/keras-team/keras-cv/discussions/52#discussioncomment-2053458

Paper : https://arxiv.org/abs/2201.09792

ConvMixer is Similar to MLP-Mixer. MLP-Mixer separates mixing of spatial and channel dimensions, by applying an MLP across spatial dimension and then an MLP across the channel dimension (spatial MLP replaces the ViT attention and channel MLP is the FFN of ViT).

ConvMixer, an extremely simple model that is similar in spirit to the ViT and the even-more-basic MLP-Mixer in that it operates directly on patches as input, separates the mixing of spatial and channel dimensions, and maintains equal size and resolution throughout the network. In contrast, however, the ConvMixer uses only standard convolutions to achieve the mixing steps. Despite its simplicity, ConvMixer outperforms the ViT, MLP-Mixer

Capture

@tanzhenyu @LukeWood @bhack @ianstenbit

### CODE REVIEW

File: keras_cv/models/convmixer.py [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+    classes=None,
+    weights=None,
+    input_shape=(None, None, 3),
+    input_tensor=None,
+    pooling=None,
+    classifier_activation="softmax",
+    name="ConvMixer_512_16",
+    **kwargs,
+):
+    """Instantiates the ConvMixer_512_16 architecture."""
```
Comment (ianstenbit) [inline keras_cv/models/convmixer.py:338]: It looks like these docstrings for specific versions of the architecture are missing the details from the docstring for the base model method.

Please add complete docstrings for these methods (the approach using `setattr` in e.g. DenseNet is a good example of how to do this without a ton of duplicate code)

**Review by ianstenbit — CHANGES_REQUESTED** [review]
Thank you!

I will kick off a training run using one of these models to validate performance compared to the paper-claimed results.

File: keras_cv/models/convmixer.py (hunk 1 above)
Comment (IMvision12) [reply keras_cv/models/convmixer.py]: Done!

**Comment by IMvision12** [comment]
Didn't mean to remove @LukeWood from the review

**Review by ianstenbit — APPROVED** [review]
Thanks! 
My training run is at 72.5% top-1 accuracy after 80 epochs for the `512/16` variant. (Paper-claimed result is 73.76). It also has 5.4M params, which matches the paper.

I'll merge this and add pre-trained weights soon!
Thanks for this contribution :)

**Comment by IMvision12** [comment]
Hey, @ianstenbit I just want to know what are hardware requirements to train ImageNet Dataset. Coz I don't think I can train it on google collab or Kaggle.

**Comment by ianstenbit** [comment]
> Hey, @ianstenbit I just want to know what are hardware requirements to train ImageNet Dataset. Coz I don't think I can train it on google collab or Kaggle.
> 
> I am working on a new architecture and want to train the new model on ImageNet Dataset.

You're most likely not able to train using ImageNet on Colab or Kaggle. @DavidLandup0 has had some success traniing with ImageNette in those environments, though (for testing new architectures)

**Comment by ianstenbit** [comment]
GCB test failure is unrelated -- I'll send a separate PR for that.

Thank you @IMvision12!

**Comment by IMvision12** [comment]
Thanks @DavidLandup0 for pointing out these things  i will open a PR fixing all these :)

**Comment by LukeWood** [comment]
> Thanks @DavidLandup0 for pointing out these things I will open a PR fixing all these :).
> 
> Also, when we talk about existing PatchingAndEmbedding layers, are we referring about Vit's PatchingAndEmbedding Layer?

Can we open an issue to track these?  Thanks!

**Comment by IMvision12** [comment]
> > Thanks @DavidLandup0 for pointing out these things I will open a PR fixing all these :).
> > Also, when we talk about existing PatchingAndEmbedding layers, are we referring about Vit's PatchingAndEmbedding Layer?
> 
> Can we open an issue to track these? Thanks!

Sure!
